tests/scripts/generate_tls13_compat_tests.py

```diff
@@ -270,6 +273,12 @@ def cmd(self):
 
         if self._named_groups:
             named_groups = ','.join(self._named_groups)
+            if self._is_hrr:
```

Comment:
For HRR test on client side we only need the first client curve and the server curve, thus wouldn't it possible to remove that and use `add_named_groups` in generate_compat_hrr_test after the client object creation to add the server curve to the client list of curves ?

Reply:
Originally I just use add_named_groups() to add the test curves, but the test will fail in one condition: -sigalgs need one group which is different from both client and server. So I have to list the curves needed here.